### ISSUE TITLE
Function calls in foreach break UnusedVariable sniff

### ISSUE THREAD

**Issue by rakelley** [slevomat/coding-standard#572]
If the left side of a `foreach` is a function/method call, the nested parentheses breaks handling of `ignoreUnusedValuesWhenOnlyKeysAreUsedInForeach`.  `UnusedVariableSniff::isValueInForeachAndErrorIsIgnored` fetches the immediately preceding parentheses, which in this case is for the function rather than the block, and returns false because the variable pointer is after the closing parentheses.

```php
        // Fails sniff
        foreach (\range(0, 5) as $key => $value) {
            \var_dump($key);
        }

        // Passes sniff
        $numbers = \range(0, 5);
        foreach ($numbers as $otherKey => $otherValue) {
            \var_dump($otherKey);
        }
```

Tested on `4.8.6`

**Comment by kukulich** [comment]
Thank you for your report.

Fixed in https://github.com/slevomat/coding-standard/commit/7b401c237697f0d3a8e447627fa52725f9c5ecd6
